vcd/datasource_vcd_nsxt_app_port_profile.go

```diff
@@ -105,28 +105,33 @@ func datasourceVcdNsxtAppPortProfileRead(_ context.Context, d *schema.ResourceDa
 	nsxtManagerId := d.Get("nsxt_manager_id").(string)
 
 	queryParams := url.Values{}
+
+	var contextId string
 	switch {
-	// For `TENANT` scope Org and VDC or the specified `context_id` matter. It would set _context
-	// filter to be searching for App Port Profiles in specific context
-	case strings.EqualFold(scope, types.ApplicationPortProfileScopeTenant):
-		contextId, err := pickAppPortProfileContextFilterByPriority(vcdClient, d, contextIdFieldValue)
+	// context_id is the preferred method of providing context
+	case contextIdFieldValue != "":
+		contextId = contextIdFieldValue
+	// if vdc attribute is set, use that, as it is also available for every user
+	case vdcName != "":
+		_, vdc, err := vcdClient.GetOrgAndVdcFromResource(d)
 		if err != nil {
-			return diag.Errorf("error identifying correct context filter: %s", err)
+			return diag.Errorf("failed to get VDC from resource: %s", err)
 		}
-		queryParams.Add("filter", fmt.Sprintf("name==%s;scope==%s;_context==%s", name, scope, contextId))
-	// For PROVIDER scoped App Port Profiles context_id of Network Provider can be specified
-	case strings.EqualFold(scope, types.ApplicationPortProfileScopeProvider) && contextIdFieldValue != "":
-		queryParams.Add("filter", fmt.Sprintf("name==%s;scope==%s;_context==%s", name, scope, contextIdFieldValue))
-	// Deprecated field 'nsxt_manager_id' can be specified as context for PROVIDER scoped App Port Profiles
-	case strings.EqualFold(scope, types.ApplicationPortProfileScopeProvider) && nsxtManagerId != "":
-		queryParams.Add("filter", fmt.Sprintf("name==%s;scope==%s;_context==%s", name, scope, nsxtManagerId))
+		contextId = vdc.Vdc.ID
+	// nsxt_manager_id can only be used by sysorg admin
+	case nsxtManagerId != "":
+		if !vcdClient.Client.IsSysAdmin {
+			return diag.Errorf("Only System administrators can provide NSX-T Manager ID")
+		}
+		contextId = nsxtManagerId
+	// if none of previous values are set, don't provide context (usable if only one nsxt manager is used)
 	default:
-		// For "SYSTEM" or "PROVIDER" scoped Application Port Profiles context can be ignored.
-		// * For "SYSTEM" this is correct behavior
-		// * For "PROVIDER" it can match App Port Profiles when multiple NSX-T Managers are
-		// configured, but this is left for backwards compatibility
-		//
-		// TODO V4 - remove support for PROVIDER scope without `context_id` field
+		contextId = ""
+	}
+
+	if contextId != "" {
```

Comment:
Maybe add a small comment about this use case? Something in the lines of what you've added to docs' field description:
https://github.com/vmware/terraform-provider-vcd/pull/1065/files#diff-38a1cf12478b001e028f2fd38ecb2416dabe92ecd3328ee59f702407b387e9baR70

Reply:
Updated.